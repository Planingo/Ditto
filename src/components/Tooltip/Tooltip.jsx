import React from 'react';
import PropTypes from 'prop-types';
import { Tooltip as AntdTooltip } from 'antd';
import './tooltip.less'

/**
 * 
 * # ðŸš§ _Composant en construction_ ðŸš§
 * 
 * # TODO
 * 
 * ## Description
 * 
 * Documentations andt du composant : https://ant.design/components/tooltip/
 * 
 */
export const Tooltip = ({children, ...props }) => {
  return (
      <AntdTooltip {...props}>{children}</AntdTooltip>
  );
};

Tooltip.propTypes = {
  /**
  * TODO property
  */
  title: PropTypes.node,
  /**
   * TODO property
  */
  children: PropTypes.node,
  /**
  * TODO property
  */
  align: PropTypes.object,
  /**
   * TODO property
  */
  arrowPointAtCenter: PropTypes.bool,
  /**
  * TODO property
  */
  autoAdjustOverflow: PropTypes.bool,
  /**
   * TODO property
  */
  color: PropTypes.string,
  /**
  * TODO property
  */
  defaultVisible: PropTypes.bool,
  /**
   * TODO property
  */
  destroyTooltipOnHide: PropTypes.bool,
  /**
  * TODO property
  */
  getPopupContainer: PropTypes.func,
  /**
   * TODO property
  */
  mouseEnterDelay: PropTypes.number,
  /**
  * TODO property
  */
  mouseLeaveDelay: PropTypes.number,
  /**
   * TODO property
  */
  overlayClassName: PropTypes.string,
  /**
  * TODO property
  */
  overlayStyle: PropTypes.object,
  /**
   * TODO property
  */
  overlayInnerStyle: PropTypes.object,
  /**
  * TODO property
  */
  placement: PropTypes.oneOf(['top', 'left', 'right', 'bottom', 'topLeft', 'topRight', 'bottomLeft', 'bottomRight', 'leftTop', 'leftBottom', 'rightTop', 'rightBottom']),
  /**
   * TODO property
  */
  trigger: PropTypes.oneOf(['hover', 'focus', 'click', 'contextMenu']),
  /**
  * TODO property
  */
  visible: PropTypes.bool,
  /**
   * TODO property
  */
  zIndex: PropTypes.number,
  /**
   * TODO property
  */
  onVisibleChange: PropTypes.func,               
};
