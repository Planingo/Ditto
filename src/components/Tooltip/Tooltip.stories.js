import React from 'react';

import { Tooltip } from './Tooltip';

export default {
  title: 'Components/Tooltip',
  component: Tooltip,
  argTypes: {
    placement: { 
      control: {
        type: 'select',
        options: ['top', 'left', 'right', 'bottom', 'topLeft', 'topRight', 'bottomLeft', 'bottomRight', 'leftTop', 'leftBottom', 'rightTop', 'rightBottom'],
      },
      table: {
        defaultValue: {
          summary: 'bottom'
        }
      }
    },
    trigger: { 
      control: {
        type: 'select',
        options: ['hover', 'focus', 'click', 'contextMenu'],
      },
      table: {
        defaultValue: {
          summary: 'hover'
        }
      }
    },
  },
  args: {
    children: "Minerva McGonagall",
    title: <p>C'est le pr√©nom et le nom</p>,
  }
};

const TooltipGlobal = (args) => <Tooltip {...args} />;

export const Defaut = TooltipGlobal.bind({});