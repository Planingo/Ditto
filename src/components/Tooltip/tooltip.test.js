import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render, waitFor, screen } from '@testing-library/react';
import React from 'react';
import {Tooltip} from './Tooltip';

describe('Tooltip component', () => {
  const props = {
    children: <p>Minerva McGonagall</p>,
    title: <p data-testid="tooltip">C'est le pr√©nom et le nom</p>,
  };

  const tooltip = (
    <Tooltip {...props} />
  );
});