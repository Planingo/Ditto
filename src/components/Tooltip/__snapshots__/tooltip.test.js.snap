// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component Snapshot tooltip 1`] = `
<body>
  <div>
    <span>
      Minerva McGonagall
    </span>
  </div>
</body>
`;
